connection.ts

```diff
@@ -158,12 +158,25 @@ export class RedisConnection implements Connection {
 
   async #connect(retryCount: number) {
     try {
-      const dialOpts: Deno.ConnectOptions = {
-        hostname: this.hostname,
-        port: parsePortLike(this.port),
```

Comment:
I thought `RedisConnectOptions` and `RedisConnectionOptions` could be merged.

```typescript
// connection.ts
interface BaseRedisConnectionOptions {
  db?: number;
  password?: string;
  username?: string;
  name?: string;
  /**
   * @default 10
   */
  maxRetryCount?: number;
  backoff?: Backoff;
  /**
   * When this option is set, a `PING` command is sent every specified number of seconds.
   */
  healthCheckInterval?: number;
}

interface RedisConnectionOptionsTCP extends BaseRedisConnectionOptions {
  tls?: boolean;
  hostname: string;
  port?: number | string;
}

interface RedisConnectionOptionsUnix extends BaseRedisConnectionOptions {
  path: string;
}

export type RedisConnectionOptions =
  | RedisConnectionOptionsTCP
  | RedisConnectionOptionsUnix;

// redis.ts
/** @deprecated */
export type RedisConnectOptions = RedisConnectionOptions;
```

Then, the constructor of `RedisConnection` could be changed as follows:

```diff
- constructor(
-   hostname: string,
-   port: number | string,
-   private options: RedisConnectionOptions,
- ) {
+ constructor(private options: RedisConnectionOptions) {
```

Reply:
Good idea, I got those mixed up at least once while working on this: 2ebea60d1545e12fc73eeb2f4cfcc740c48cf2f4

While I was at it, I've also removed assigning `"unix"` to the `port` and the socket path to `hostname`. Those are private vars, but it might trip someone up in the future: 72364508001175a09ae9da048207c7f167d3e0fb